openvmm/openvmm_entry/src/lib.rs

```diff
@@ -1534,28 +1576,36 @@ fn disk_open(disk_cli: &DiskCliKind, read_only: bool) -> anyhow::Result<Resource
                 _ => {}
             }
 
-            Resource::new(disk_backend_resources::LayeredDiskHandle {
-                layers: vec![
-                    SqliteDiskLayerHandle {
-                        dbhd_path: path.display().to_string(),
-                        format_dbhd: create.then_some(
-                            disk_backend_resources::layer::SqliteDiskLayerFormatParams {
-                                logically_read_only: false,
-                                len: None,
-                            },
-                        ),
-                    }
-                    .into_resource()
-                    .into(),
-                    DiskLayerHandle(disk_open(disk, true)?)
-                        .into_resource()
-                        .into(),
-                ],
-            })
+            layers.push(LayerOrDisk::Layer(
+                SqliteDiskLayerHandle {
+                    dbhd_path: path.display().to_string(),
+                    format_dbhd: create.then_some(
+                        disk_backend_resources::layer::SqliteDiskLayerFormatParams {
+                            logically_read_only: false,
+                            len: None,
+                        },
+                    ),
+                }
+                .into_resource()
+                .into(),
+            ));
+            disk_open_inner(disk, true, layers)?;
         }
-    };
-
-    Ok(disk_type)
+        DiskCliKind::AutoCacheSqlite { key, disk } => {
+            layers.push(LayerOrDisk::Layer(DiskLayerDescription {
+                read_cache: true,
+                write_through: false,
+                layer: SqliteAutoCacheDiskLayerHandle {
+                    cache_path: std::env::var("OPENVMM_AUTO_CACHE_PATH")
```

Comment:
having this "floating" env var defined here seems a bit unfortunate...

could we route this parameter though `clap` instead? e.g: via a `#[clap(env)] auto_cache_path`?

Reply:
It's awkward to put it in a `clap(env)` because you really want the failure point to know what the name of the env var is to report it to the user. But I moved the lookup into the parsing of the `DiskCliKind`, which I think is better.